src/seb/registered_models/cohere_models.py

```diff
@@ -4,54 +4,74 @@
 
 
 import logging
-from collections.abc import Sequence
 from functools import partial
 from typing import Any
 
 import torch
 
 import seb
+from seb.interfaces.task import Task
 from seb.registries import models
 
 logger = logging.getLogger(__name__)
 
 
 class CohereTextEmbeddingModel(seb.Encoder):
-    def __init__(self, model_name: str) -> None:
+    def __init__(self, model_name: str, sep: str = " ") -> None:
         self.model_name = model_name
-
-    @staticmethod
-    def create_sentence_blocks(
-        sentences: Sequence[str],
-        block_size: int,
-    ) -> list[Sequence[str]]:
-        sent_blocks: list[Sequence[str]] = []
-        for i in range(0, len(sentences), block_size):
-            sent_blocks.append(sentences[i : i + block_size])
-        return sent_blocks
+        self.sep = sep
 
     def get_embedding_dim(self) -> int:
-        v = self.encode(["get emb dim"])
+        v = self._embed(["get emb dim"], input_type="classification")
         return v.shape[1]
 
-    def encode(
-        self,
-        sentences: Sequence[str],
-        batch_size: int = 32,  # noqa: ARG002
-        embed_type: str = "classification",
-        **kwargs: Any,  # noqa: ARG002
-    ) -> torch.Tensor:
-        import cohere  # type: ignore
+    def _embed(self, sentences: list[str], input_type: str) -> torch.Tensor:
+        import cohere
 
         client = cohere.Client()
         response = client.embed(
             texts=list(sentences),
             model=self.model_name,
-            input_type=embed_type,
+            input_type=input_type,
         )
-
         return torch.tensor(response.embeddings)
 
+    def encode(
+        self,
+        sentences: list[str],
+        batch_size: int = 32,  # noqa: ARG002
+        *,
+        task: Task,
+        **kwargs: Any,  # noqa: ARG002
+    ) -> torch.Tensor:
+        if task.task_type == "Classification":
+            input_type = "classification"
+        elif task.task_type == "Clustering":
+            input_type = "clustering"
+        else:
+            input_type = "search_document"
```

Comment:
retrieval should be search document. 

Should it be the default as well?

Reply:
Retrieval tasks use asymmetric encoding anyways, so hopefully this function will never run in that case. As for the default, I don't see why not. The two tasks that do not have specific input types in Cohere are STS and BitextMining, both of which (in my humble judgement) are most similar to retrieval. What do you think, @KennethEnevoldsen?

Reply:
I think this is fine. Just wanted to make sure we had considered it.

Reply:
No problem, just wanted to make sure